corehq/form_processor/abstract_models.py

```diff
@@ -192,6 +192,31 @@ def get_properties_in_api_format(self):
             # all custom properties go here
         }.items())
 
+    @property
+    @memoized
+    def parent(self):
```

Comment:
It looks like this change might conflict (or be trying to do the same thing) as https://github.com/dimagi/commcare-hq/pull/10961/files#diff-79b08116b1c7db03795d235524c2a26eR189. might be worth reconciling the two approaches